CrossPare/src/main/java/de/ugoe/cs/cpdp/execution/AbstractCrossProjectExperiment.java

```diff
@@ -181,6 +181,7 @@ public void run() {
                     versionCount++;
                     continue;
                 }
+                SetUniqueList<SoftwareVersion> trainversionOriginalSet = SetUniqueList.setUniqueList(trainversionSet);

```

Comment:
are you sure this is a deep copy and the subsequent changes do not affect this copy? I have not checked myself, so I really do not know. please just double check because this would be annoying. Easiest way would be to just use Undersampling or Oversampling and see if the values are as expected.

Reply:
a different solution may be just to pull up the "makeSingleVersionSet" for the copy and do this here instead of in line 225, because that should be a deep copy if I remember correctly.

Reply:
Yes, I checked it with oversampling beforehand. Anyway, your idea seems to be even cleaner.